book/src/css.md

```diff
@@ -0,0 +1,264 @@
+
+<span class="filename">Filename: listings/css/1/main.rs</span>
+
+```rust ,no_run,noplayground
+{{#rustdoc_include ../listings/css/1/main.rs:main}}
+```
+
+When we now run the app, we notice that our button *and* the "close" button are red.
+Probably not what we wanted, but that is what we typed.
+We did not specify for which button the rule should apply, so it was applied to both of them.
+
+<div style="text-align:center"><img src="img/css_1.png"/></div>
+
+>The `GtkInspector` comes in quite handy (not only) when playing with CSS.
+>Make sure that the window of your app is focused and press <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>D</kbd>.
+> A window will pop up which lets you browse and even manipulate the state of your app.
+> Open the `CSS` view and override the button color with the following snippet.
+>```css
+>button {
+>  color: blue;
+>}
+>```
+
+## Classes
+
+[Class selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Class_selectors) are one way to limit the amount of elements a CSS rule applies to.
+Let us look at few different scenarios where they are involved.
+
+
+### Classes Applied by GTK
+
+GTK adds style classes to many of its nodes, often depending on a certain condition.
+In the case of the node [`button`](https://gtk-rs.org/gtk4-rs/stable/latest/docs/gtk4/struct.Button.html#css-nodes), the style class `.text-button` will be added when the content of the corresponding widget is just a label.
+That is why we create a new CSS rule which only applies to `button` nodes with the style class `text_button`.
+
+
+<span class="filename">Filename: listings/css/2/style.css</span>
+
+```css
+{{#rustdoc_include ../listings/css/2/style.css}}
+```
+
+<div style="text-align:center"><img src="img/css_2.png"/></div>
+
+### Classes Applied Manually
+
+If we want to apply different rules to nodes which got the same style classes added, we will have to think of a different strategy.
+For that we add our own CSS class with [`add_css_class`](../docs/gtk4/prelude/trait.WidgetExt.html#tymethod.add_css_class).
+
+<span class="filename">Filename: listings/css/3/main.rs</span>
+
+```rust ,no_run,noplayground
+{{#rustdoc_include ../listings/css/3/main.rs:buttons}}
+```
+
+And we create a CSS rule that applies to `button` nodes with the style class `button_1`.
+
+<span class="filename">Filename: listings/css/3/style.css</span>
+
+```css
+{{#rustdoc_include ../listings/css/3/style.css}}
+```
+
+This way, only the first button gets colored red.
+If you want to make sure that CSS rules only apply to a hand-selected set of nodes, adding your own style classes is your best bet.
+Just make sure that you do not choose a name for your style class that GTK either adds itself or provides CSS rules for.
+This brings us to the next point.
+
+<div style="text-align:center"><img src="img/css_3.png"/></div>
+
+### CSS Rules Provided by GTK
+
+Certain styles are common enough that GTK provides its own CSS rules for.
```

Comment:
```suggestion
Certain styles are common enough that GTK provides preset CSS rules for them in form of style classes.
```

Reply:
Isn't style class used in this section to mean css class?

Reply:
Adwaita is a stylesheet. `.suggested-action` and `.text-button` are style classes.

Reply:
Yes, I was confused